ScrollerMac shouldn't use hardcoded NSControlSize

lukewarlow opened the pull request:
#### 78b4149d570a3d45b62b4d6312c7692d2f7f18e4
<pre>
ScrollerMac shouldn&apos;t use hardcoded NSControlSize
<a href="https://bugs.webkit.org/show_bug.cgi?id=258752">https://bugs.webkit.org/show_bug.cgi?id=258752</a>

Reviewed by Simon Fraser.

This patch moves the function to convert from ScrollbarWidth to NSControlSize into ScrollTypesMac.h.
This is then used inside of ScrollerMac instead of a hard coded NSControlSizeRegular.
ScrollerPairMac currently returns a hardcoded ScrollbarWidth::Auto, so this patch has no behaviour changes.

* Source/WebCore/page/scrolling/mac/ScrollerMac.mm:
(WebCore::ScrollerMac::attach):
(WebCore::ScrollerMac::updateScrollbarStyle):
* Source/WebCore/page/scrolling/mac/ScrollerPairMac.h:
* Source/WebCore/page/scrolling/mac/ScrollerPairMac.mm:
(WebCore::ScrollerPairMac::scrollbarWidthStyle const):
* Source/WebCore/platform/mac/ScrollTypesMac.h:
(WebCore::nsControlSizeFromScrollbarWidth):
* Source/WebCore/platform/mac/ScrollbarThemeMac.mm:
(WebCore::ScrollbarThemeMac::registerScrollbar):
(WebCore::ScrollbarThemeMac::scrollbarThickness):
(WebCore::scrollbarWidthToNSControlSize): Deleted.

Canonical link: <a href="https://commits.webkit.org/265691@main">https://commits.webkit.org/265691@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0f2d4a2d1267071fa9bd52521791552275247c85

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/11658 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/11858 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/12225 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/13300 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/11103 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/11677 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/14246 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/11838 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/13972 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/11822 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/12665 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/9886 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/13723 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/9957 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/10582 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/17716 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/11032 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/10736 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/13913 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/11135 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/9184 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/10312 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/2799 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/14592 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/10992 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on Source/WebCore/page/scrolling/mac/ScrollerPairMac.h:
```diff
@@ -72,6 +72,7 @@ class ScrollerPairMac : public ThreadSafeRefCountedAndCanMakeThreadSafeWeakPtr<S
 
     FloatSize visibleSize() const;
     bool useDarkAppearance() const;
+    ScrollbarWidth scrollbarWidthStyle() const;
```
Is this actually called anywhere?

lukewarlow replied:
My bad it's meant to be called inside of ScrollerMac. I'll fix that now

lukewarlow replied:
It is now

lukewarlow committed:
ScrollerMac shouldn't use hardcoded NSControlSize
https://bugs.webkit.org/show_bug.cgi?id=258752

Reviewed by Simon Fraser.

This patch moves the function to convert from ScrollbarWidth to NSControlSize into ScrollTypesMac.h.
This is then used inside of ScrollerMac instead of a hard coded NSControlSizeRegular.
ScrollerPairMac currently returns a hardcoded ScrollbarWidth::Auto, so this patch has no behaviour changes.

* Source/WebCore/page/scrolling/mac/ScrollerMac.mm:
(WebCore::ScrollerMac::attach):
(WebCore::ScrollerMac::updateScrollbarStyle):
* Source/WebCore/page/scrolling/mac/ScrollerPairMac.h:
* Source/WebCore/page/scrolling/mac/ScrollerPairMac.mm:
(WebCore::ScrollerPairMac::scrollbarWidthStyle const):
* Source/WebCore/platform/mac/ScrollTypesMac.h:
(WebCore::nsControlSizeFromScrollbarWidth):
* Source/WebCore/platform/mac/ScrollbarThemeMac.mm:
(WebCore::ScrollbarThemeMac::registerScrollbar):
(WebCore::ScrollbarThemeMac::scrollbarThickness):
(WebCore::scrollbarWidthToNSControlSize): Deleted.

Canonical link: https://commits.webkit.org/265691@main
